crates/egui/src/widgets/slider.rs

```diff
@@ -539,15 +563,37 @@ impl<'a> Slider<'a> {
 
         if let Some(pointer_position_2d) = response.interact_pointer_pos() {
             let position = self.pointer_position(pointer_position_2d);
-            let new_value = if self.smart_aim {
-                let aim_radius = ui.input(|i| i.aim_radius());
+            let mut new_value = if self.smart_aim {
+                let aim_radius = ui.input().aim_radius();
                 emath::smart_aim::best_in_range_f64(
                     self.value_from_position(position - aim_radius, position_range.clone()),
                     self.value_from_position(position + aim_radius, position_range.clone()),
                 )
             } else {
                 self.value_from_position(position, position_range.clone())
             };
+            if !self.snap_values.is_empty() {
+                let mut closest_snap = 0.0;
+                let norm_value = normalized_from_value(new_value, self.range.clone(), &self.spec);
+                let mut closest_distance = f64::INFINITY;
+                for snap in self.snap_values.iter().copied() {
+                    let snap_norm_value =
+                        normalized_from_value(snap, self.range.clone(), &self.spec);
+                    let distance = (norm_value - snap_norm_value).abs();
+                    if distance < closest_distance {
+                        closest_snap = snap;
+                        closest_distance = distance;
+                    } else {
+                        break;
+                    }
+                }
+
+                // Divide aim_radius by the granularity we want
+                if self.snap_values_only || closest_distance < ui.input().aim_radius() as f64 / 15.0 {
+                    new_value = closest_snap;
+                }
```

Comment:
The docs aren't very good around `position_from_value` etc, but you should measure the distances in ui space, NOT in the space of the slider values.

So do this:

```suggestion
                // We use ui points as the unit for the measurements:
                let mut closest_snap = 0.0;
                let new_value_pos = self.position_from_value(new_value, position_range.clone());
                let mut closest_distance = f64::INFINITY;
                for snap_value in self.snap_values.iter().copied() {
                    let snap_pos = self.position_from_valuee(snap_value, self.range.clone());
                    let distance = (new_value_pos - snap_pos).abs();
                    if distance < closest_distance {
                        closest_snap = snap;
                        closest_distance = distance;
                    } else {
                        break;
                    }
                }

                // Divide aim_radius by the granularity we want
                if self.snap_values_only || closest_distance < ui.input().aim_radius() {
                    new_value = closest_snap;
                }
```